### PR TITLE
Modernize and run eslint on tools/acorn-optimizer.js. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: tools/acorn-optimizer.js [hunk 1]
```diff
@@ -53,13 +54,15 @@ function visitChildren(node, c) {
...
   }
-  for (var key in node) {
-    var child = node[key];
+  for (const key in node) {
     // Check for a child.
-    if (!maybeChild(child)) {
-      // Check for an array of children.
-      if (Array.isArray(child)) {
-        child.forEach(maybeChild);
+    if (Object.prototype.hasOwnProperty.call(node, key)) {
```
Comment (aheejin) [inline tools/acorn-optimizer.js:59]: Does lint auto-change code in LHS to RHS by adding these lines? (Not very familiar with eslint myself)

File: tools/acorn-optimizer.js [hunk 2]
```diff
@@ -287,13 +299,13 @@ function hasSideEffects(node) {
 // as they appear (like ArrowFunctionExpression). Instead, we do a conservative
 // analysis here.
 
-function JSDCE(ast, aggressive) {
+function runJSDCE(ast, aggressive) {
```
Comment (aheejin) [inline tools/acorn-optimizer.js:302]: Does eslint rename functions?

File: tools/acorn-optimizer.js (hunk 1 above)
Comment (sbc100) [reply tools/acorn-optimizer.js:59]: No, the linter complains that when you iterate through an object with `for in` you need to include an inner check for hasOwnProperty:

Here is the rule: https://eslint.org/docs/rules/guard-for-in

File: tools/acorn-optimizer.js (hunk 2 above)
Comment (sbc100) [reply tools/acorn-optimizer.js:302]: There is a rule that says that any function that starts with a Capital letter must be a constructor function (and this isn't one).

**Review by aheejin — APPROVED** [review]
Didn't check each change in detail, but LGTM if this is what eslint tells to do. By the way, does eslint fix things by itself, or we need fix it manually? How can I run eslint on a specific js file based on our configuration?

**Comment by sbc100** [comment]
Yes, eslint can fix some/all of the issue it finds automatically, but I have no yet chosen to use that feature, preferring to do it myself so that I internalize the rules and can understand the changes being made.  So this change, for example, was fully hand rolled.

To run eslint you can just run `eslint <filename>.js` if is lint is in your path... or `npx esling <filename>.js` if its not.   Or even `./node_modules/.bin/eslint <filename>.js` (assuming your have already run `npm install`).

**Pull request merged**